src/core/server/saved_objects/service/lib/repository.test.js

```diff
@@ -781,14 +801,42 @@ describe('SavedObjectsRepository', () => {
           id: 'three',
         };
         const objects = [obj1, obj, obj2];
-        const response = getMockBulkCreateResponse(objects);
+        const response = getMockBulkCreateResponse([obj1, obj2]);
         callAdminCluster.mockResolvedValue(response); // this._writeToCluster('bulk', ...)
         const result = await savedObjectsRepository.bulkCreate(objects);
         expect(callAdminCluster).toHaveBeenCalledTimes(1);
         expect(result).toEqual({
           saved_objects: [expectSuccessResult(obj1), expectError(obj), expectSuccessResult(obj2)],
         });
       });
+
+      it(`a deserialized saved object`, async () => {
+        // Test for fix to https://github.com/elastic/kibana/issues/65088 where
+        // we returned raw ID's when an object without an id was created.
+        const namespace = 'myspace';
+        const response = getMockBulkCreateResponse([obj1, obj2], namespace);
+        callAdminCluster.mockResolvedValueOnce(response); // this._writeToCluster('bulk', ...)
+
+        // Bulk create one object with id unspecified, and one with id specified
+        const result = await savedObjectsRepository.bulkCreate([{ ...obj1, id: undefined }, obj2], {
+          namespace,
+        });
+
+        // Assert that both raw docs from the ES response are deserialized
+        expect(serializer.rawToSavedObject).toHaveBeenNthCalledWith(1, {
+          ...response.items[0].create,
+          _id: expect.stringMatching(/myspace:config:[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}/),
+        });
+        expect(serializer.rawToSavedObject).toHaveBeenNthCalledWith(2, response.items[1].create);
+
+        // Assert that ID's are deserialized to remove the type and namespace
+        expect(result.saved_objects[0].id).toEqual(
+          expect.not.stringMatching(/config|index-pattern|myspace/)
+        );
+        expect(result.saved_objects[1].id).toEqual(
+          expect.not.stringMatching(/config|index-pattern|myspace/)
+        );
```

Comment:
Can't we use the uuid regexp (`[0-9a-f]{8}-([0-9a-f]{4}-){3}[0-9a-f]{12}`) as done L828  instead of asserting against the names of the types/spaces used in the test suite? I think that's a little more explicit?

Reply:
`obj2`'s id is `logstash-*` so I initially thought I'll try to keep these assertions the same. But I agree, a regexp would be better, so I'll change the first match to use a regexp and the second match to match `obj2.id`.